### PR TITLE
[adobepass] Add AT&T U-Verse MSO

### PR DESCRIPTION
**Pull request by jkap**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Added AT&T U-Verse as a supported adobepass MSO.

I'm not sure what order `MSO_INFO` should be in, if `ATT` should be placed somewhere else in the dict I can move it.

### CODE REVIEW

File: youtube_dl/extractor/adobepass.py [hunk 1]
```diff
@@ -20,6 +20,11 @@
 
 
 MSO_INFO = {
+    'ATT': {
+        'name': 'AT&T U-Verse',
+        'username_field': 'nameBox',
+        'password_field': 'pwdBox'
```
Comment (dstftw) [inline youtube_dl/extractor/adobepass.py]: Browser does not send such fields.

**Comment by dstftw** [comment]
For this to get merged you must provide account credentials.

**Comment by jkap** [comment]
@dstftw what's the preferred way to get account login to y'all?

File: youtube_dl/extractor/adobepass.py (hunk 1 above)
Comment (mso0919) [reply youtube_dl/extractor/adobepass.py]: It sends "userid" and "password" instead.
`<input type="text" name="userid" id="nameBox" value="" maxlength="50" size="50" class="slidTxtbox" title="Userid" style="color: rgb(170, 170, 170);">`
`<input type="text" placeholder="Password" name="password" id="pwdBox" value="" maxlength="50" size="50" class="slidTxtbox" autocomplete="off" title="Password">`

**Comment by DreamStatic** [comment]
Has this been implemented yet? I am getting an error when using ATT as the mso stating "youtube-dl: error: Unsupported TV Provider, use --ap-list-mso to get a list of supported TV Providers". When I scan through the list I do not see anything related to ATT U-Verse. Is it under a different name?

**Comment by sjeble** [comment]
Any update on this?

**Comment by benhager** [comment]
Is anyone still working on this? I would love to get this working... Anything I can do to help? We have an active AT&T U-verse account if it needs testing.

**Comment by gthrift** [comment]
I can provide uverse credentials if we can get this implemented.  How do I share?

**Comment by benhager** [comment]
For what it's worth, adding the initially suggested code to the MSO list works for me without issue. I'm not sure what the complaint was by @dstftw and @mso0919 
`    'ATT': {
        'name': 'AT&T U-Verse',
        'username_field': 'nameBox',
        'password_field': 'pwdBox'
    },`

The text 'nameBox' and 'pwdBox' are returned as the **id** of the two fields and not the **name** attribute. Could that be causing the confusion? Hope this can be cleared up as this would be great to have as a part of the next release.

Let me know if I can help!

**Comment by benhager** [comment]
FWIW, I aslo test the suggestion by @mso0919 to use this in the list and it seems to work without issue as well.

`'ATT': {
        'name': 'AT&T U-Verse',
        'username_field': 'userid',
        'password_field': 'password'
    },`

Not sure how you'd all like to proceed, but I would recommend merging one of them at least!

**Pull request closed without merging**